lib/transformer.cc

```diff
@@ -81,91 +88,130 @@ void RotateTransformer::TransformCanvas::SetPixel(int x, int y, uint8_t red, uin
   }
 }
 
-int RotateTransformer::TransformCanvas::width() const {
+void RotateTransformer::TransformCanvas::SetImage(uint8_t *rawdata, int length, bool isBGR)
```

Comment:
Make sure to use the curly brace rules used in the rest of the file (open curly braces after parenthesis etc).

This could probably be formulated a bit more efficiently to reduce calculations

   * store width and height in a local variable beforehand so that we don't have to call a virtual method in each loop iteration.
   * it might be slightly faster to swap the loops and have outer loop y and inner loop x, and in the inner loop rawdata += 3 after setting the pixel. Thus, we don't have to calculate the offset every time
   * if you are concenred about performance, maybe putting the `if (isBGR) ...` outside and have two slightly different versions of the loop could avoid one branch in the inner loop. Might be worthwhile to benchmark, it might not be too much of a difference.